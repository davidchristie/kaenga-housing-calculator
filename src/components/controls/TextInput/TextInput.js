import PropTypes from 'prop-types'
import React, { Component } from 'react'
import {
  ControlLabel,
  FormControl,
  FormGroup,
  HelpBlock
} from 'react-bootstrap'

class TextInput extends Component {
  render () {
    const {
      input: { onChange, value },
      label,
      meta: { error },
      type
    } = this.props
    return (
      <FormGroup>
        <ControlLabel>{label}</ControlLabel>
        <FormControl onChange={onChange} value={value} type={type} />
        {error && <HelpBlock>{error}</HelpBlock>}
      </FormGroup>
    )
  }
}

TextInput.defaultProps = {
  input: {},
  meta: {}
}

TextInput.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.any
  }).isRequired,
  label: PropTypes.string,
  meta: PropTypes.shape({
    error: PropTypes.string
  })
}

export default TextInput
