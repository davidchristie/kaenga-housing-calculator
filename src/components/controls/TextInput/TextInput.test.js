import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import TextInput from './TextInput'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<TextInput />)).toMatchSnapshot()
})
