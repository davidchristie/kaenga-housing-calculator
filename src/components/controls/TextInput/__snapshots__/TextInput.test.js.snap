exports[`test matches snapshot 1`] = `
<FormGroup
  bsClass="form-group">
  <ControlLabel
    bsClass="control-label"
    srOnly={false} />
  <FormControl
    bsClass="form-control"
    componentClass="input"
    input={Object {}}
    meta={Object {}}
    onChange={undefined}
    value={undefined} />
</FormGroup>
`;
